src/kv/apply_changes.h

```diff
@@ -59,15 +59,8 @@ namespace ccf::kv
 
     for (auto it = changes.begin(); it != changes.end(); ++it)
     {
-      bool changeset_has_writes = it->second.changeset->has_writes();
-      if (changeset_has_writes)
-      {
-        has_writes = true;
-      }
-      if (changeset_has_writes || track_read_versions)
-      {
-        it->second.map->lock();
-      }
+      has_writes |= it->second.changeset->has_writes();
```

Comment:
I think we ought to drop track_read_versions altogether from the call, that's a vestige of aft.